File: backend/app/api/v1/tasks.py
"""Task API routes."""

import uuid

from fastapi import APIRouter, Depends, Query

from app.api.dependencies import get_task_service
from app.schemas.task import TaskResponse, TaskUpdate
from app.services.task_service import TaskService

router = APIRouter(prefix="/tasks", tags=["tasks"])


@router.get("", response_model=list[TaskResponse])
async def list_tasks(
    owner: str | None = Query(None, description="Filter by owner name"),
    status: str | None = Query(None, description="Filter by task status"),
    priority: str | None = Query(None, description="Filter by priority level"),
    service: TaskService = Depends(get_task_service),
) -> list[TaskResponse]:
    """List tasks with optional AND-combined filters."""
    filters: dict[str, str] = {}
    if owner is not None:
        filters["owner"] = owner
    if status is not None:
        filters["status"] = status
    if priority is not None:
        filters["priority"] = priority

    return await service.list_tasks(filters or None)


@router.patch("/{task_id}", response_model=TaskResponse)
async def update_task(
    task_id: uuid.UUID,
    body: TaskUpdate,
    service: TaskService = Depends(get_task_service),
) -> TaskResponse:
    """Partially update a task."""
    return await service.update_task(task_id, body)

File: frontend/src/api/tasks.ts
import type { Task, TaskFilters, TaskUpdate } from '../types'
import { get, patch, getBlob } from './client'

export async function listTasks(filters?: TaskFilters): Promise<Task[]> {
  return get<Task[]>('/tasks', filters as Record<string, string | undefined>)
}

export async function updateTask(id: string, data: TaskUpdate): Promise<Task> {
  return patch<Task>(`/tasks/${id}`, data)
}

export async function exportTasksCsv(filters?: TaskFilters): Promise<Blob> {
  return getBlob('/tasks/export', filters as Record<string, string | undefined>)
}
